ads/feature_store/mixin/oci_feature_store.py

```
@@ -4,20 +4,60 @@
 # Copyright (c) 2023 Oracle and/or its affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
 
+import logging
+import os
+logger = logging.getLogger(__name__)
 from ads.common.oci_mixin import OCIModelMixin
 import oci.feature_store
-import os
+import yaml
+
+
+try:
+    from yaml import CDumper as dumper
+    from yaml import CLoader as loader
+except:
+    from yaml import Dumper as dumper
+    from yaml import Loader as loader
+
+try:
+    from odsc_cli.utils import user_fs_config_loc, FsTemplate
+except ImportError:
+    pass
 
 
 class OCIFeatureStoreMixin(OCIModelMixin):
+    __mod_time = 0
+    __template: "FsTemplate" = None
+    FS_SERVICE_ENDPOINT = "fs_service_endpoint"
+    SERVICE_ENDPOINT = "service_endpoint"
+
     @classmethod
     def init_client(
         cls, **kwargs
     ) -> oci.feature_store.feature_store_client.FeatureStoreClient:
-        # TODO: Getting the endpoint from authorizer
-        fs_service_endpoint = os.environ.get("OCI_FS_SERVICE_ENDPOINT")
+
+        default_kwargs:dict = cls._get_auth().get("client_kwargs",{})
+
+        fs_service_endpoint = kwargs.get(cls.FS_SERVICE_ENDPOINT, None) or kwargs.get(cls.SERVICE_ENDPOINT, None) or \
+                              default_kwargs.get(cls.FS_SERVICE_ENDPOINT, None)
+
+        if not fs_service_endpoint:
+            try:
+                mod_time = os.stat(user_fs_config_loc()).st_mtime
+                if mod_time > cls.__mod_time:
+                    with open(user_fs_config_loc()) as ccf:
+                        cls.__template = FsTemplate(yaml.load(ccf, Loader=loader))
+                    cls.__mod_time = mod_time
+            except NameError:
+                logger.info("%s", "Feature store configuration helpers are missing. "
+                            "Support for reading service endpoint from config file is disabled")
+            except FileNotFoundError:
+                pass
```

can we log a warning in this case instead of just eating the exception?

---

sure...I will log it at info level since it is ok for the file to be not their